extensions/binding/org.eclipse.smarthome.binding.yahooweather/src/main/java/org/eclipse/smarthome/binding/yahooweather/internal/connection/YahooWeatherConnection.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.yahooweather.internal.connection;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.commons.io.IOUtils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation for Yahoo Weather url connection.
+ * 
+ * @author Christoph Weitkamp - Changed use of caching utils to ESH ExpiringCacheMap
+ * 
+ */
+public class YahooWeatherConnection {
+
+    private final Logger logger = LoggerFactory.getLogger(YahooWeatherConnection.class);
+
+    private final static String WEBSERVICE_URL = "https://query.yahooapis.com/v1/public/yql?format=json";
+
+    public String getResponseFromQuery(String query) {
+        try {
+            final URL url = new URL(
+                    WEBSERVICE_URL + "&q=" + URLEncoder.encode(query, StandardCharsets.UTF_8.toString()));
+            final URLConnection connection = url.openConnection();
+            return IOUtils.toString(connection.getInputStream());
```

Comment:
Do you need to close the input or output stream?

WRT:
Invoking the close() methods on the InputStream or OutputStream of an URLConnection after a request may free network resources associated with this instance, unless particular protocol specifications specify different behaviours for it.